dts: arm: stm32f4: Add quadspi node

KimBP opened the pull request:
By some reason qspi support has never been added to stm32f4xx

Think this is what is needed

erwango commented:
@KimBP Have you been able to test this ?
Beides, would you be able to enable qspi in one F4 board upstream ?

KimBP commented:
Unfortunately haven't tested yet. 
I have access to a STM32F412G_Disco board so it ought be possible. 
I'll enable and test it on that board and then get back with an update to this PR

kim-esco committed:
boards: stm32f412g_disco: Enable qspi-nor flash

The qspi-nor flash is enabled.
Can be tested with e.g. drivers/spi_flash sample

Missing documentation of support of SPI and I2C added

Signed-off-by: Kim Bøndergaard <kim.boendergaard@escoglobal.com>

KimBP commented:
@erwango : stm32f412g_Disco board now updated. 
Changes verified with sample drivers_spi_flash:

*** Booting Zephyr OS build zephyr-v2.6.0-1515-g82d8dcd102d0  ***

JEDEC QSPI-NOR SPI flash testing
==========================

Test 1: Flash erase
Flash erase succeeded!

Test 2: Flash write
Attempting to write 4 bytes
Data read matches data written. Good!!

erwango commented:
Thanks @KimBP!

erwango commented on dts/arm/st/f4/stm32f4.dtsi:
```diff
@@ -470,6 +470,17 @@
+			compatible = "st,stm32-qspi";
+			#address-cells = <0x1>;
+			#size-cells = <0x0>;
+			reg = <0xa0001000 0x34>;
```
Nitpick: Length is not used today, but for now convention is to set it as 0x400 for all devices.

erwango commented on dts/arm/st/f4/stm32f4.dtsi:
```diff
@@ -470,6 +470,17 @@
+			#size-cells = <0x0>;
+			reg = <0xa0001000 0x34>;
+			interrupts = <92 0>;
+			clocks = <&rcc STM32_CLOCK_BUS_AHB3 0x2>;
```
```suggestion
			clocks = <&rcc STM32_CLOCK_BUS_AHB3 0x00000002>;
```

erwango commented on boards/arm/stm32f412g_disco/stm32f412g_disco.dts:
```diff
@@ -123,3 +123,20 @@
 	status = "okay";
 };
+
+&quadspi {
```
Please update board documentation accordingly

erwango reviewed (changes requested):
Thanks for this contribution.
Some minor changes, otherwise LGTM

cfriedt commented:
@KimBP  - please address @FRASTM 's comments.

KimBP commented on dts/arm/st/f4/stm32f4.dtsi:
```diff
@@ -470,6 +470,17 @@
 			label = "DMA_2";
 		};
+
+		quadspi: quadspi@a0001000 {
```
OK - wasn't aware of that - but have had the thought. 
I'm currently interested in stm32f412xg but think we should put it where it covers as many devices as possible.
Does it exist in all stm32f412 variants?
stm32f412.dtsi by the way includes stm32f411 (which again includes stm32f401).
I.e putting it into. stm32f411.dtsi could be an option too ?

KimBP reviewed (commented):
@FRASTM : Will do. Originally it was set to 0x34, but erwango claimed the convention is to set it to 0x400. Don't know why

cfriedt commented:
@erwango, @FRASTM - is this change ready to go? I wasn't sure if all concerns were addressed, but it has both approvals atm

KimBP committed:
dts: arm: stm32f412: Add quadspi node

register address 0xa0001000 length 0x400
Interrupt 92
Clockbit 1

Signed-off-by: Kim Bøndergaard <kim@fam-boendergaard.dk>

KimBP commented:
@cfriedt I think I've followed up on all issues being raised
